Editor/HPUIInteractorConeRayAnglesEditor.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEditor;
+using ubco.ovilab.HPUI.Interaction;
+using UnityEngine.XR.Hands;
+
+namespace ubco.ovilab.HPUI.Editor
+{
+    [CanEditMultipleObjects]
+    [CustomEditor(typeof(HPUIInteractorConeRayAngles), true)]
+    public class HPUIInteractorConeAnglesEditor: UnityEditor.Editor
+    {
+        private HPUIInteractorConeRayAngles t;
+
+        private float offsetValue;
+        private bool applyToAll;
+        private bool overwriteThisAsset;
+        private XRHandJointID selectedPhalange = XRHandJointID.IndexDistal;
+
+        // storing all baseline thresholds for first ray per phalange
+        // assuming that offsets are applied uniformly per phalange
+        private Dictionary<XRHandJointID, float> baselineThresholds = new Dictionary<XRHandJointID, float>();
+        // storing previous applied offset
+        private Dictionary<XRHandJointID, float> cachedOffsets = new Dictionary<XRHandJointID, float>();
+
+        private const string DONT_ASK_FOR_OVERRIDE_EDITOR_PREF_KEY = "ubco.ovilab.HPUI.Components.ConeThresholdsEditor.DontAskForOverride";
+
+        protected void OnEnable()
+        {
+            t = target as HPUIInteractorConeRayAngles;
+            if (!t) return;
+            foreach ((XRHandJointID phalange, List<HPUIInteractorRayAngle> rays) in t.ActiveFingerAngles)
+            {
+                if (!baselineThresholds.ContainsKey(phalange) && rays.Count > 0)
+                {
+                    baselineThresholds[phalange] = rays.Select(ray => ray.RaySelectionThreshold).ToList()[0];
+                }
+
+                if (cachedOffsets.ContainsKey(phalange)) continue;
+                float initialOffset =
+                    rays.Count > 0 ? rays[0].RaySelectionThreshold - baselineThresholds[phalange] : 0f;
+                cachedOffsets[phalange] = initialOffset;
+            }
+
+            Undo.undoRedoPerformed += OnUndoRedoEvent;
+        }
+
+        protected void OnDisable()
+        {
+            Undo.undoRedoPerformed -= OnUndoRedoEvent;
+        }
+
+        public override void OnInspectorGUI()
+        {
+            EditorGUILayout.LabelField("Apply Ray Selection Threshold Offset", EditorStyles.boldLabel);
+            offsetValue = EditorGUILayout.FloatField("Offset", offsetValue);
+            applyToAll = EditorGUILayout.Toggle("Apply to All", applyToAll);
+            overwriteThisAsset = EditorGUILayout.Toggle("Overwrite This Asset", overwriteThisAsset);
+
+            EditorGUI.BeginDisabledGroup(applyToAll);
+            selectedPhalange = (XRHandJointID)EditorGUILayout.EnumPopup("Phalange", selectedPhalange);
+            EditorGUI.EndDisabledGroup();
+
+            if (GUILayout.Button(overwriteThisAsset ? "Overwrite this Asset and Apply Offset" : "Create Copy and Apply Offset"))
+            {
+                Undo.RecordObject(t, "Apply RaySelectionThreshold Offset");
+
+                if (applyToAll)
```

Comment:
Consider restructuring this a bit:

```csharp
string saveName;
bool applyOffsets = false;
HPUIInteractorConeRayAngles assetModying;
if (overwriteThisAsset) 
{
    assetModying = t;
    // get display dialog output (applyOffsets)
}
else
{
    assetModying = //instantiate
    applyOffsets = true;
    // get saveName
}

if (applyOffsets)
{
    if (applyToAll)
    {
    ....
    }
    else
    {
    ....
    }
    if (!overwriteThisAsset)
    {
        AssetDatabase.CreateAsset(assetModying, saveName);
    }
    EditorUtility.SetDirty(assetModying);
    AssetDatabase.SaveAssets();
    AssetDatabase.Refresh();
}
```

Separate the UI logic from the domain logic where possible